session.go

```diff
@@ -44,15 +41,31 @@ type Pool struct {
 	closed bool
 }
 
+// Logger is used by mgosession to log information about sessions.
+type Logger interface {
+	// Debug logs a message at debug level.
+	Debug(message string)
+	// Info logs a message at info level.
+	Info(message string)
+}
+
 // NewPool returns a session pool that maintains a maximum
 // of maxSessions sessions available for reuse.
-func NewPool(ctx context.Context, s *mgo.Session, maxSessions int) *Pool {
+//
+// All the sessions will be coped (with Session.Copy) from s.
+//
+// The logger is used to log informational messages about the pool
+// and may be nil if no logging is required.
+func NewPool(logger Logger, s *mgo.Session, maxSessions int) *Pool {
```

Comment:
I'm not sure. Should we pass context.Context to every single method and function that we define because one day it might be of use? Should we add context.Context to Close and Reset too?

My current thinking is not - we can break compatibility if we need context in the future.